model_signing/serialize_test.py

```diff
@@ -9,12 +9,46 @@
 
 
 # Utility functions.
-def create_folder(name: str) -> Path:
+def create_empty_folder(name: str) -> Path:
     p = os.path.join(os.getcwd(), testdata_dir, name)
     os.makedirs(p)
     return Path(p)
 
 
+def create_random_folders(name: str) -> (Path, int, [Path], [Path]):
```

Comment:
Question: why not use [`tempfile.mkdtemp`](https://docs.python.org/3/library/tempfile.html)?

Reply:
you mea as the location of the folder? Yeah I can do that. Maybe I should update other functions that create files as well then?

Reply:
Yeah, I'm thinking of having calls to `mkdtemp`/`mkstemp` to create the directories/files (and act as context managers) and then a separate utility to write random bytes to a file. Then we could have the test be something like

```python
with random_directory() as dir:
  with random_directory(dir) as dir1:
     with random_file(dir1) as f:
        generate_random_file_contents(f)  # /dir/dir1/f actually
        sign(dir)
# now we are guaranteed that the directory is removed
```

This could also be done later, if needed. I just think it will be more readable and helps with cleaning up.